### PR TITLE
Add build_manual and build_vignettes to install functions

### LINKED ISSUE #353: Argument for building vignettes in install_*()
It would be very convenient to have an explicit argument for building vignettes for the family of `install_*()` functions. This would be similar to what was available via the `build_vignettes` argument in the pre-`remotes` (before version 2.0.0) version of `devtools::install_*()`, which still survives in the current [`devtools::install()`](https://github.com/r-lib/devtools/blob/ab6c878d9ec3657fc08f07fa4389e23374617d01/R/install.R#L64).

Although one can manually specify `build_opts = c("--no-resave-data", "--no-manual")`, an explicit argument is particularly useful when describing e.g. how to install a package from GitHub in its README. One typically wants vignettes to be available to the user installing the package, especially if the package is not on CRAN. In this regard, an explicit 
```r
remotes::install_github("my/pkg", build_vignettes = TRUE)
```
is a lot more informative for the user than
```r
remotes::install_github("my/pkg", build_opts = c("--no-resave-data", "--no-manual"))
```
where it is not evident this is not about the specified `biuld_opts` but about "excluding the exclusion of vignettes from the pkg build".

Or would you recommend for such cases to even use `build_opts = NULL` for a fully-fledged build?

### PR DESCRIPTION
**Pull request by jimhester**
Fixes #353

### CODE REVIEW

File: tests/testthat/test-install.R [hunk 1]
```diff
@@ -26,8 +26,9 @@ test_that("safe_build_package fails appropriately with pkgbuild", {
   on.exit(unlink(out, recursive = TRUE))
   opts <- c("--no-resave-data", "--no-manual", "--no-build-vignettes")
   expect_error(
-    safe_build_package(test_path("invalidpkg"), build_opts = opts , out, quiet = TRUE, use_pkgbuild = TRUE),
-    "System command error"
+    safe_build_package(test_path("invalidpkg"), build_opts = opts, build_manual = FALSE, build_vignettes = FALSE, out, quiet = TRUE, use_pkgbuild = TRUE),
+    "System command error",
+    class = "system_command_status_error"
```
Comment (jimhester) [inline tests/testthat/test-install.R]: This just silences a warning from the newer testthat, as this is a classed error message.

**Comment by jimhester** [comment]
An alternative implementation would be to have the `install_*()` functions call `normalize_build_opts()` themselves and just pass the normalized build_opts onwards. I thought it was a bit nicer to keep that as an implementation detail in `install()`, though it required more work to pass the arguments correctly.

**Comment by jimhester** [comment]
I think I would prefer to do this in remotes rather than putting wrappers in devtools, as we also discussed.

File: install-github.R [hunk 2]
```diff
@@ -3784,7 +3830,25 @@ safe_install_packages <- function(...) {
   )
 }
 
-safe_build_package <- function(pkgdir, build_opts, dest_path, quiet, use_pkgbuild = !is_standalone() && pkg_installed("pkgbuild")) {
+normalize_build_opts <- function(build_opts, build_manual, build_vignettes) {
+  if (!isTRUE(build_manual)) {
+    build_opts <- c(build_opts, "--no-manual")
```
Comment (jennybc) [inline install-github.R]: Just for parallelism with `setdiff()`, should this `c()` and the one for `build_vignettes` be `union()` and then you could ditch the `unique()`?

**Comment by jennybc** [comment]
Sad news: I fetched this PR and, while there are no obvious signs of trouble with `install_github("r-lib/fs", build_vignettes = TRUE)`, I don't seem to have vignettes when it's done. fs is a good test case.

File: install-github.R (hunk 2 above)
Comment (jimhester) [reply install-github.R]: That is a good suggestion, yeah

**Comment by jimhester** [comment]
Hmm, it seems to work fine for me, this is what I see, the important bit to look for is

> ─  installing the package to build vignettes
✔  creating vignettes (25.1s)

```
> install_github("r-lib/fs", build_vignettes = TRUE)
Downloading GitHub repo r-lib/fs@master
   checking for file ‘/private/var/folders/dt/r5s12t392tb5sk181j3gs4zw0000gn
✔  checking for file ‘/private/var/folders/dt/r5s12t392tb5sk181j3gs4zw0000gn
/T/Rtmp8ZUbVN/remotes168402bb6da04/r-lib-fs-1555cd0/DESCRIPTION’
─  preparing ‘fs’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  installing the package to build vignettes
✔  creating vignettes (25.1s)
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘fs_1.3.1.9000.tar.gz’

Installing package into ‘/Users/jhester/.temp’
(as ‘lib’ is unspecified)
* installing *source* package ‘fs’ ...
** libs
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c dir.cc -o dir.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c error.cc -o error.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c file.cc -o file.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c fs.cc -o fs.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c id.cc -o id.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c link.cc -o link.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c path.cc -o path.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c utils.cc -o utils.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c unix/getmode.cc -o unix/
getmode.o
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./
libuv/include -I. -pthread -I"/Users/jhester/Library/R/3.5/library/Rcpp/incl
ude" -I/usr/local/include   -fPIC  -Wall -g -O2  -c RcppExports.cpp -o RcppE
xports.o
cp -p -f libuv/m4/lt_obsolete.m4 libuv/m4/lt~obsolete.m4
(cd libuv \
                && touch aclocal.m4 \
                && touch -r aclocal.m4 configure Makefile.in \
        && CC="clang" CFLAGS="-Wall -g -O2  -fPIC " AR="ar" RANLIB="ranlib"
LDFLAGS="-L/usr/local/lib" ./configure --quiet)
rm: cannot remove 'conftest.dSYM': Is a directory
rm: cannot remove 'conftest.dSYM': Is a directory
rm: cannot remove 'conftest.dSYM': Is a directory
rm: cannot remove 'conftest.dSYM': Is a directory
make --directory=libuv \
                HAVE_DTRACE=0
  CC       src/libuv_la-fs-poll.lo
  CC       src/libuv_la-inet.lo
  CC       src/libuv_la-threadpool.lo
  CC       src/libuv_la-version.lo
  CC       src/libuv_la-uv-common.lo
  CC       src/unix/libuv_la-async.lo
  CC       src/unix/libuv_la-core.lo
  CC       src/unix/libuv_la-dl.lo
  CC       src/unix/libuv_la-fs.lo
  CC       src/unix/libuv_la-getaddrinfo.lo
  CC       src/unix/libuv_la-getnameinfo.lo
  CC       src/unix/libuv_la-loop-watcher.lo
  CC       src/unix/libuv_la-loop.lo
  CC       src/unix/libuv_la-pipe.lo
  CC       src/unix/libuv_la-poll.lo
  CC       src/unix/libuv_la-process.lo
  CC       src/unix/libuv_la-signal.lo
  CC       src/unix/libuv_la-stream.lo
  CC       src/unix/libuv_la-tcp.lo
  CC       src/unix/libuv_la-thread.lo
  CC       src/unix/libuv_la-timer.lo
  CC       src/unix/libuv_la-tty.lo
  CC       src/unix/libuv_la-udp.lo
  CC       src/unix/libuv_la-bsd-ifaddrs.lo
  CC       src/unix/libuv_la-darwin.lo
  CC       src/unix/libuv_la-darwin-proctitle.lo
  CC       src/unix/libuv_la-fsevents.lo
  CC       src/unix/libuv_la-kqueue.lo
  CC       src/unix/libuv_la-proctitle.lo
  CCLD     libuv.la
clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_look
up -single_module -multiply_defined suppress -L/Library/Frameworks/R.framewo
rk/Resources/lib -L/usr/local/lib -o fs.so dir.o error.o file.o fs.o id.o li
nk.o path.o utils.o unix/getmode.o RcppExports.o ./libuv/.libs/libuv.a -pthr
ead -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,Cor
eFoundation
installing to /Users/jhester/.temp/fs/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (fs)
Adding ‘fs_1.3.1.9000.tgz’ to the cache
> browseVignettes(package = "fs")
```

fs might be a tricky case because it is a devtools dependency via usethis, as not everything might get unloaded cleanly...

**Comment by jennybc** [comment]
OK I just had a successful fs experiment. I sure *thought* I had installed remotes from this PR, but I must have only checked out the branch. Sorry for the hiccup.

**Comment by jennybc** [comment]
> Sad news: I fetched this PR and, while there are no obvious signs of trouble with `install_github("r-lib/fs", build_vignettes = TRUE)`, I don't seem to have vignettes when it's done. fs is a good test case.

The silent failure of my first attempt is presumably yet another example of `...` quietly absorbing `build_vignettes = TRUE` and then doing nothing with it.

File: NEWS.md [hunk 3]
```diff
@@ -9,6 +9,9 @@
 * Fix bug in internal `parse_deps()` where test of valid comparison operators
 failed due to trailing whitespaces in DESCRIPTION fields (@LiNk-NY, #366)
 
+* `install_*()` functions gain `build_manual` and `build_vignette` arguments
```
Comment (jennybc) [inline NEWS.md]:
```suggestion
* `install_*()` functions gain `build_manual` and `build_vignettes` arguments
```

**Pull request merged**